### PR TITLE
Show a size estimate for the png export button

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
This PR adds a rough order of magnitude size estimate for the png export button:

image

This is an attempt to warn a user what they're getting into in the event someone produces a massive CFG :)

### CODE REVIEW

File: static/panes/cfg-view.ts [hunk 1]
```diff
@@ -61,6 +61,8 @@ type Coordinate = {
 const DZOOM = 0.1;
 const MINZOOM = 0.1;
 
+const EST_COMPRESSION_RATIO = 0.022;
```
Comment (jeremy-rifkin) [inline static/panes/cfg-view.ts:64]: From an experimental run, should be an over-estimate

File: static/panes/cfg-view.ts [hunk 2]
```diff
@@ -77,6 +79,33 @@ function attrs(attributes: Record<string, string | number | null>) {
         .join(' ');
 }
 
+function special_round(x: number) {
```
Comment (jeremy-rifkin) [inline static/panes/cfg-view.ts:82]: I would not be surprised if there is a better way to do this and I am greatly overcomplicating things 😅

File: views/templates/panes/cfg.pug [hunk 3]
```diff
@@ -9,7 +9,7 @@
       .dropdown-menu
         button.btn.btn-sm.btn-light.dropdown-item.export-png(title="Export graph" aria-label="Export graph as an image")
           span.fas.fa-image
-          span.hideable Export PNG
+          span.hideable Export PNG <span class="estimated-export-size"></span>
```
Comment (jeremy-rifkin) [inline views/templates/panes/cfg.pug:12]: Not sure if this is idiomatic pug

Comment (mattgodbolt) [reply views/templates/panes/cfg.pug:12]: I think indenting another line under works?

Comment (mattgodbolt) [reply views/templates/panes/cfg.pug:12]:
```suggestion
          span.hideable Export PNG
            span.estimated-export-size
```

not sure it's better, but it's more pug-gy

Comment (jeremy-rifkin) [reply views/templates/panes/cfg.pug:12]: Oh yeah let's do that

**Pull request merged**